tracer/src/Datadog.Trace/Agent/Transports/ApiWebRequest.cs

```diff
@@ -46,5 +55,103 @@ public async Task<IApiResponse> PostAsync(ArraySegment<byte> bytes, string conte
                 return new ApiWebResponse((HttpWebResponse)exception.Response);
             }
         }
+
+        public async Task<IApiResponse> PostAsync(params MultipartFormItem[] items)
+        {
+            if (items is null)
+            {
+                ThrowHelper.ThrowArgumentNullException(nameof(items));
+            }
+
+            Log.Debug<int>("Sending multipart form request with {Count} items.", items.Length);
+
+            _request.Method = "POST";
+            _request.ContentType = "multipart/form-data; boundary=" + Boundary;
+
+            using (var requestStream = await _request.GetRequestStreamAsync().ConfigureAwait(false))
+            {
+                // Write form request using the boundary
+                var boundaryBytes = _boundarySeparatorInBytes ??= Encoding.ASCII.GetBytes(BoundarySeparator);
+                var trailerBytes = _boundaryTrailerInBytes ??= Encoding.ASCII.GetBytes(BoundaryTrailer);
+
+                // Write each MultipartFormItem
+                foreach (var item in items)
+                {
+                    byte[] headerBytes = null;
+
+                    // Check name is not null (required)
+                    if (item.Name is null)
+                    {
+                        Log.Warning("Error encoding multipart form item name is null. Ignoring item");
+                        continue;
+                    }
+
+                    // Ignore the item if the name contains ' or "
+                    if (item.Name.Contains("\"") || item.Name.Contains("'"))
```

Comment:
Maybe should be this? (if so, there's the version below)
```suggestion
                    if (item.Name.Contains('\"', StringComparison.Ordinal) || item.Name.Contains('\'', StringComparison.Ordinal))
```

Reply:
That's not available in all targets, so I'll change it to an `IndexOf`

Reply:
changed in [bf8a1ae](https://github.com/DataDog/dd-trace-dotnet/pull/2962/commits/bf8a1aec483504e1c01a37ddec207ffa36f8e872)